esrally/client/asynchronous.py

```diff
@@ -290,8 +290,13 @@ def __init__(self, *args, **kwargs):
         # The async client is used in the hot code path and we use customized overrides (such as that we don't
         # parse response bodies in some cases for performance reasons, e.g. when using the bulk API).
         self._verified_elasticsearch = True
+        self._serverless = False
+
+        # this isn't always available because any call to self.options() doesn't pass any custom args
+        # to the constructor
+        # https://github.com/elastic/rally/issues/1673
         if distribution_version:
-            self.distribution_version = versions.Version.from_string(distribution_version)
+            self.distribution_version = distribution_version
         else:
             self.distribution_version = None
```

Comment:
Why don't we override the options() method ourselves like this instead?

```
def options(self, *, **kwargs):
   newself = super().options(**kwargs)
   newself.distribution_version = self.distribution_version
   newself._serverless = self._serverless
   return newself
```

It seems like it would solve both distribution_version being None (which is very annoying) and needing RallyAsyncElasticsearchServerless (which is not that bad but that I'd prefer to avoid if possible).

Reply:
Thank you! Implemented in https://github.com/elastic/rally/pull/1738/commits/abf43207e1ebde6899876dcc3ebadf5998e7eab1, testing with some proper benchmarks now.